js/src/components/free-listings/configure-product-listings/checkErrors.test.js

```diff
@@ -0,0 +1,352 @@
+/**
+ * Internal dependencies
+ */
+import checkErrors from './checkErrors';
+
+function toRates( ...tuples ) {
+	return tuples.map( ( [ countryCode, rate ] ) => ( {
+		countryCode,
+		rate,
+	} ) );
+}
+
+function toTimes( ...tuples ) {
+	return tuples.map( ( [ countryCode, time ] ) => ( {
+		countryCode,
+		time,
+	} ) );
+}
+
+describe( 'checkErrors', () => {
+	describe( 'Shipping rates', () => {
+		let flatShipping;
+		let manualShipping;
+
+		beforeEach( () => {
+			flatShipping = { shipping_rate: 'flat' };
+			manualShipping = { shipping_rate: 'manual' };
+		} );
+
+		it( 'When the type of shipping rate is an invalid value, should not pass', () => {
+			// Not set yet
+			let errors = checkErrors( {}, [], [], [] );
+
+			expect( errors ).toHaveProperty( 'shipping_rate' );
+			expect( errors.shipping_rate ).toMatchSnapshot();
+
+			// Invalid value
+			errors = checkErrors( { shipping_rate: true }, [], [], [] );
+
+			expect( errors ).toHaveProperty( 'shipping_rate' );
+			expect( errors.shipping_rate ).toMatchSnapshot();
+
+			// Invalid value
+			errors = checkErrors( { shipping_rate: 'automatic' }, [], [], [] );
+
+			expect( errors ).toHaveProperty( 'shipping_rate' );
+			expect( errors.shipping_rate ).toMatchSnapshot();
+		} );
+
+		it( 'When the type of shipping rate is a valid value, should pass', () => {
+			// Selected flat
+			let errors = checkErrors( flatShipping, [], [], [] );
+
+			expect( errors ).not.toHaveProperty( 'shipping_rate' );
+
+			// Selected manual
+			errors = checkErrors( manualShipping, [], [], [] );
+
+			expect( errors ).not.toHaveProperty( 'shipping_rate' );
+		} );
+
+		describe( 'For flat type', () => {
+			function createFreeShipping( threshold ) {
+				return {
+					...flatShipping,
+					offers_free_shipping: true,
+					free_shipping_threshold: threshold,
+				};
+			}
+
+			it( `When there are any selected countries' shipping rates is not set, should not pass`, () => {
```

Comment:
:nail_care: 
I'm not an English expert, but, I'm not sure if that's grammatically correct.

```suggestion
			it( `When there are any selected countries with shipping rates not set, should not pass`, () => {
```